🐛 Fixed error handling in async helpers

ErisDS opened the pull request:
This is a 3 part fix:

**Commit 1** is a global fix that should ensure any error thrown inside of any async helper will always be properly handled.

When this fix is applied, an error will be logged and an async helper like {{prev_post}} will render an error in development mode, or nothing in production mode.

**Commit 2** adds error handling to the prev/next helper, and unifies the way prev_next is treated so it behaves much more closely to the get helper.

When this fix is applied, an error will be logged and an async helper like {{prev_post}} will not render if there is an error, regardless of mode.

**Commit 3** downgrades the errors thrown by img_url to a logged warning.

When this fix is applied, a **warning** will be logged and async helpers like {{prev_post}} are unaffected.

closes #8703

kirrg001 commented:
I've tested the changes with a broken theme, they all look correct to me 👍 Is there a reason why the PR is WIP?

ErisDS commented:
No, sorry! I added the WIP whenever I or ened the PR and it only had the first commit, and forgot to take it off when I updated the PR.

I did this, but recommend testing the positive cases - that non broken themes still work correctly also :)
